library/modules/Military.cpp

```diff
@@ -289,3 +295,116 @@ void Military::updateRoomAssignments(int32_t squad_id, int32_t civzone_id, df::s
         }
     }
 }
+
+static bool remove_soldier_entity_link(df::historical_figure* hf, df::squad* squad)
+{
+    int32_t start_year = -1;
+    for (size_t i = 0; i < hf->entity_links.size(); i++)
+    {
+        auto link = strict_virtual_cast<df::histfig_entity_link_squadst>(hf->entity_links[i]);
```

Comment:
`strict_virtual_cast` can be slow. it's good to guard it with a check of  `link->getType()`